src/AutoMapper/ConstructorMap.cs

```diff
@@ -1,81 +1,83 @@
-﻿using AutoMapper.QueryableExtensions;
-using AutoMapper.QueryableExtensions.Impl;
-
-namespace AutoMapper
-{
-    using System;
-    using System.Collections.Generic;
-    using System.Linq;
-    using System.Linq.Expressions;
-    using System.Reflection;
-    using Execution;
-
-    public class ConstructorMap
-    {
-        private static readonly DelegateFactory DelegateFactory = new DelegateFactory();
-        private readonly Lazy<LateBoundParamsCtor> _runtimeCtor;
-        public ConstructorInfo Ctor { get; private set; }
-        internal ConstructorParameterMap[] CtorParams { get; }
-
-        public ConstructorMap(ConstructorInfo ctor, ConstructorParameterMap[] ctorParams)
-        {
-            Ctor = ctor;
-            CtorParams = ctorParams;
-
-            _runtimeCtor = new Lazy<LateBoundParamsCtor>(() => DelegateFactory.CreateCtor(ctor, CtorParams));
-        }
-
-        private static readonly IExpressionResultConverter[] ExpressionResultConverters =
-        {
-            new MemberGetterExpressionResultConverter(),
-            new ExpressionBasedResolverResultConverter(),
-            new MemberResolverExpressionResultConverter(),
-            new NullSubstitutionExpressionResultConverter()
-        };
-
-        public bool CanResolve
+﻿using AutoMapper.QueryableExtensions;
+using AutoMapper.QueryableExtensions.Impl;
+
+namespace AutoMapper
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Linq.Expressions;
+    using System.Reflection;
+    using Configuration;
+    using Execution;
+    using Mappers;
+
+    public class ConstructorMap
+    {
+        private bool _sealed;
+        private static readonly DelegateFactory DelegateFactory = new DelegateFactory();
+        private readonly Lazy<LateBoundParamsCtor> _runtimeCtor;
+        public ConstructorInfo Ctor { get; }
+        internal ConstructorParameterMap[] CtorParams { get; }
+
+        public ConstructorMap(ConstructorInfo ctor, ConstructorParameterMap[] ctorParams)
+        {
+            Ctor = ctor;
+            CtorParams = ctorParams;
+
+            _runtimeCtor = new Lazy<LateBoundParamsCtor>(() => DelegateFactory.CreateCtor(ctor, CtorParams));
+        }
+
+        private static readonly IExpressionResultConverter[] ExpressionResultConverters =
+        {
+            new MemberGetterExpressionResultConverter(),
+            new MemberResolverExpressionResultConverter(),
+        };
+
+        public bool CanResolve => CtorParams.All(param => param.CanResolve);

```

Comment:
Yes, it's [enumerators](https://cloud.githubusercontent.com/assets/4517428/13640021/ec66519a-e61b-11e5-9053-c230f928715d.png).